### PR TITLE
refactor docker compose and remove add data

### PR DESCRIPTION
**Pull request by sylvaingaudan**
- split docker compose in one docker compose file per service + network + volume
- add a "hat" docker compose that includes the others
- simplify README.md
- remove the versions from README and docker files. Only one place for the versions : `env.sh`. Having multiple places containing the version is error prone.
- in the docker file, references to the services are made relative and not absolute, it simplifies the configuration
- only one basemap. Should move to protomap soon.
- network becomes `arlas-net` instead of `esnet` which was not suitable
- use `-p arlas_exploration_stack` for grouping the microservices in one project
- add release for tagging and nginx

### CODE REVIEW

File: scripts/build_docker.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    PLATFORM=$3
+fi
+
+echo "build for ${PLATFORM} platform"
+
+build__docker (){
+    IMAGE=$1
+    VERSION=$2
+    echo "Building the image $IMAGE:${VERSION} (${BUILD_ARG})"
+    docker build ${BUILD_ARG} --platform ${PLATFORM} -f ${IMAGE}/Dockerfile -t gisaia/${IMAGE}:${VERSION} -t gisaia/${IMAGE}:latest  -t ${IMAGE}:${VERSION} -t ${IMAGE}:latest .
```
Comment (QuCMGisaia) [inline scripts/build_docker.sh:25]: Do we always want to build the latest version of the images ? I think some dev has been done to put that as optional in other projects

File: scripts/publish_docker.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+[ -z "$1" ] && echo "Please provide the name of the compononet" && exit 1;
+[ -z "$2" ] && echo "Please provide the version" && exit 1;
+
+publish_docker (){
+    IMAGE=$1
+    VERSION=$2
+    echo "Publishing the image $IMAGE"
+    docker login
+    docker push gisaia/${IMAGE}:latest
```
Comment (QuCMGisaia) [inline scripts/publish_docker.sh]: Same as above

File: start.sh [hunk 3]
```diff
@@ -237,40 +224,35 @@ fi
...
-# Kill the spinner on any signal, including our own exit.
-trap "kill -9 $SPIN_PID" `seq 0 15`
 
 #We need to stop useless local services started because of depends_on value in docker_compose.yaml
 
 if [ "$ignore_persistence" = true ]; 
     then
-        eval "docker-compose -f $SCRIPT_DIRECTORY/docker-compose.yaml stop arlas-persistence-server"
-        eval "docker-compose -f $SCRIPT_DIRECTORY/docker-compose.yaml rm --force arlas-persistence-server"
+        eval "docker-compose -p arlas_exploration_stack  -f $SCRIPT_DIRECTORY/docker-compose-arlas-stack.yaml stop arlas-persistence-server"
```
Comment (QuCMGisaia) [inline start.sh]: Double spaces before the -f

File: start.sh [hunk 4]
```diff
@@ -289,8 +271,8 @@ if [ "$ignore_es" = true ];
             fi
             eval "sleep 5"
         done 
-        #Restart Arlas server when we are sure thar ES is UP
-        eval "docker-compose -f $SCRIPT_DIRECTORY/docker-compose.yaml restart arlas-server"
+        #Restart Arlas server when we are sure that ES is UP
```
Comment (QuCMGisaia) [inline start.sh]: Arlas -> ARLAS

**Review by QuCMGisaia — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -29,38 +20,31 @@ Usage: ./start.sh [--arlas-permissions-url] [--arlas-persistence-url] [--arlas-s
...
 
-ELASTICSEARCH options enable sniffing : false
-ELASTICSEARCH options enable SSL : false
-ELASTICSEARCH credentials :
-ARLAS ELASTICSEARCH index : .arlas
-````
-```localhost``` can be replaced by an env variable __$LOCAL_HOST__ in `.env` file
+## Services
+
+The stack contains 6 services :
```
Comment (MohamedHamouGisaia) [inline README.md]: 7 services

File: docker-compose-arlas-server.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  arlas-server:
+    image: gisaia/arlas-server:${ARLAS_SERVER_VERSION:-latest}
+    container_name: arlas-server
+    restart: always
+    depends_on:
+      elasticsearch:
+        condition: service_healthy
+    expose:
+      - "9999"
+    ports:
```
Comment (MohamedHamouGisaia) [inline docker-compose-arlas-server.yaml]: Is it necessary to keep this exposition ? we can interact with localhost:81/server no ?

File: scripts/release.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+[ -z "$1" ] && echo "Please provide the version" && exit 1;
+VERSION=$1
+
+# NGINX
+./scripts/build_docker.sh arlas-stack-nginx $VERSION "linux/amd64"
+./scripts/publish_docker.sh arlas-stack-nginx $VERSION "linux/amd64"
+
+git tag -a ${VERSION} -m "COF-NG ${VERSION}"
```
Comment (MohamedHamouGisaia) [inline scripts/release.sh]: trailing COF-NG

File: scripts/build_docker.sh (hunk 1 above)
Comment (sylvaingaudan) [reply scripts/build_docker.sh:25]: Ok, I make it optional

File: scripts/publish_docker.sh (hunk 2 above)
Comment (sylvaingaudan) [reply scripts/publish_docker.sh]: same as above

File: start.sh (hunk 3 above)
Comment (sylvaingaudan) [reply start.sh]: ok

File: start.sh (hunk 4 above)
Comment (sylvaingaudan) [reply start.sh]: ok

File: docker-compose-arlas-server.yaml (hunk 6 above)
Comment (sylvaingaudan) [reply docker-compose-arlas-server.yaml:16]: ok

File: scripts/release.sh (hunk 7 above)
Comment (sylvaingaudan) [reply scripts/release.sh]: ok

File: docker-compose-arlas-server.yaml (hunk 6 above)
Comment (sylvaingaudan) [reply docker-compose-arlas-server.yaml]: ok

File: arlas-stack-nginx/Dockerfile
Comment (mbarbet) [inline arlas-stack-nginx/Dockerfile:1]: should we not specify a version for nginx ?

**Review by mbarbet — APPROVED** [review]

**Pull request merged**